rs/artifact_pool/src/certification_pool.rs

```diff
@@ -23,9 +23,71 @@ use ic_types::{
     consensus::HasHeight,
     Height,
 };
-use prometheus::IntCounter;
+use prometheus::{labels, opts, IntCounter, IntGauge};
 use std::collections::{BTreeMap, HashSet};
 
+struct PerTypeMetrics {
```

Comment:
What was the reason that you can't re-use the `PerTypeMetrics` struct that we have for the consensus pool here?

Reply:
The `PerTypeMetrics` in consensus has methods that are hard-coded for consensus (like `new` and `update`). Re-using them here felt wrong and error-prone. I tried making PerTypeMetrics more generic, but the certification pool's unvalidated pool isn't a `HeightIndexedPool`, so I would have to create something special for that anyways. 

Alternatively, I could make `InMemoryPoolSection` (which implements `HeightIndexedPool`) abstract over both consensus and certification messages, so that the unvalidated certification pool can use it. But that would lead to some more problems.